implementations/micrometer-registry-new-relic/src/test/java/io/micrometer/newrelic/NewRelicMeterRegistryTest.java

```diff
@@ -15,372 +15,903 @@
+import static org.mockito.ArgumentMatchers.anyMap;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.CALLS_REAL_METHODS;
+import static org.mockito.Mockito.RETURNS_DEFAULTS;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
 
 /**
  * Tests for {@link NewRelicMeterRegistry}.
  *
  * @author Johnny Lim
+ * @author Galen Schmidt
  */
+@SuppressWarnings("ConstantConditions")
 class NewRelicMeterRegistryTest {
 
-    private final NewRelicConfig config = new NewRelicConfig() {
-        
-        @Override
-        public String get(String key) {
-            return null;
-        }
-        
-        @Override
-        public String accountId() {
-            return "accountId";
-        }
-        
+    private final NewRelicConfig meterNameEventTypeDisabledConfig = new MockNewRelicConfig() {
+
         @Override
-        public String apiKey() {
-            return "apiKey";
+        public boolean meterNameEventTypeEnabled() {
+            return false; // note: false is the default value
         }
-        
+
     };
-   
-    private final NewRelicConfig meterNameEventTypeEnabledConfig = new NewRelicConfig() {
-        
+
+    private final NewRelicConfig meterNameEventTypeEnabledConfig = new MockNewRelicConfig() {
+
         @Override
         public boolean meterNameEventTypeEnabled() {
-            // Previous behavior for backward compatibility
             return true;
         }
-        
-        @Override
-        public String get(String key) {
-            return null;
-        }
-        
-        @Override
-        public String accountId() {
-            return "accountId";
-        }
-        
-        @Override
-        public String apiKey() {
-            return "apiKey";
-        }
-        
+
     };
-    
+
+    private final Insights insights = Mockito.mock(Insights.class, RETURNS_DEFAULTS);
+
+    private final HttpSender httpSender = mock(HttpSender.class, CALLS_REAL_METHODS);
+
+    private final NamedThreadFactory threadFactory = new NamedThreadFactory(getClass().getSimpleName());
+
+    private final ObjectMapper mapper = new ObjectMapper();
+
     private final MockClock clock = new MockClock();
+
     private final NewRelicMeterRegistry meterNameEventTypeEnabledRegistry = new NewRelicMeterRegistry(meterNameEventTypeEnabledConfig, clock);
-    private final NewRelicMeterRegistry registry = new NewRelicMeterRegistry(config, clock);
+
+    private final NewRelicMeterRegistry meterNameEventTypeDisabledRegistry = new NewRelicMeterRegistry(meterNameEventTypeDisabledConfig, clock);
+
+    @Test
+    void eventFor_Gauge() {
+        eventFor_Gauge(this.meterNameEventTypeEnabledRegistry, ImmutableMap.<String, Object>builder()
+                .put("eventType", "myGauge")
```

Comment:
The previous test did a simple String comparison between the generated JSON and the expected value. This broke with the addition of `NewRelicEvent`, because the attributes were now stored in a Map, meaning the keys were written in an non-deterministic order in the JSON.

I've refactored the tests to use Jackson to read the JSON object into a Map, and then use AssertJ's map comparator instead. That way, the tests only care that the JSON has the correct keys, values, and types, making the tests easier to write + less prone to false failures.